src/UI/Dice/DicePlay.razor

```diff
@@ -0,0 +1,185 @@
+@attribute [MatchFor(typeof(DiceEngine), ComponentScopes.GamePlay)]
```

Comment:
General comment: nesting is pretty huge in this file. I'd probably extract some sub-components here.